### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by MartinBernstorff**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 danavis/
-.vscode/
\ No newline at end of file
+.vscode/
+*.json
+.DS*
```
Comment (KennethEnevoldsen) [inline .gitignore]: the star here is a bit odd.

File: basics.py [hunk 2]
```diff
@@ -1,53 +1,39 @@
...
+        if count % 5 == 0:
+            print(f"Processing {count}")
 
-        # KCE:alternative to token count
-        # token_count += len(doc)
+        token_count += len(doc)
             
         for token in doc:
-            token_count += 1
+            pos_counts[token.tag_] += 1 #Loving counters right here
```
Comment (KennethEnevoldsen) [inline basics.py:23]: Looks great

File: .gitignore (hunk 1 above)
Comment (KennethEnevoldsen) [reply .gitignore]: Could exclude a large group of unintended files.

Comment (KennethEnevoldsen) [reply .gitignore]: ps. th .DS_store is still on the github. Challenge: remove it ;)

File: pos_counts_extension.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import spacy
+import os
+
+def file_generator(dir: str) -> Generator[str, None, None]:
+    for filename in os.listdir(dir):
+        if "danavis" in filename:
+            filepath = os.path.join(dir, filename)
+
+            with open(filepath, "r") as f:
+                yield f.read()
```
Comment (KennethEnevoldsen) [inline pos_counts_extension.py]: copy paste. You should import from a common script. e.g. utils.py

File: pos_counts_extension.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+def get_pos_counts(doc: Doc) -> Counter:
+    token_count = 0
+    pos_counts = Counter()
+
+    token_count += len(doc)
```
Comment (KennethEnevoldsen) [inline pos_counts_extension.py]: token_count is unused?

**Review by KennethEnevoldsen — CHANGES_REQUESTED** [review]
Looks good - few minor corrections

File: .gitignore (hunk 1 above)
Comment (MartinBernstorff) [reply .gitignore]: Haha, you bastard! Thought I'd have to wrangle a lot with git here, but it's done ;-)

**Pull request merged**